app/src/androidTest/java/com/github/sdpteam15/polyevents/view/fragments/MapFragmentTest.kt

```diff
@@ -0,0 +1,141 @@
+            .check(matches(isDisplayed()))
+    }
+
+    @After
+    fun teardown() {
+        currentDatabase = FirestoreDatabaseProvider
+    }
+
+    @Test
+    fun btnIsCorrectlyDisplayedVisitorMap() {
+        Espresso.onView(ViewMatchers.withId(R.id.id_fragment_map))
+            .check(matches(isDisplayed()))
+        Espresso.onView(ViewMatchers.withId(R.id.id_heatmap))
+            .check(matches(isDisplayed()))
+        Espresso.onView(ViewMatchers.withId(R.id.id_location_button))
+            .check(matches(isDisplayed()))
+        Espresso.onView(ViewMatchers.withId(R.id.id_locate_me_button))
+            .check(matches(isDisplayed()))
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_delete_areas)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.saveNewRoute)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.removeRoute)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.addNewRoute)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.saveAreas)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_edit_area)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.addNewArea)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.acceptNewArea)
+    }
+
+    @Test
+    fun btnIsCorrectlyDisplayedRouteMap() {
+        Espresso.onView(ViewMatchers.withId(R.id.ic_home))
+            .perform(click())
+        Espresso.onView(ViewMatchers.withId(R.id.id_route_manager_button))
+            .perform(scrollTo(), click())
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_delete_areas)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.saveNewRoute)
+        BaristaVisibilityAssertions.assertDisplayed(R.id.removeRoute)
+        BaristaVisibilityAssertions.assertDisplayed(R.id.addNewRoute)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.saveAreas)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_edit_area)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.addNewArea)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.acceptNewArea)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_location_button)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_heatmap)
+        BaristaVisibilityAssertions.assertNotDisplayed(R.id.id_locate_me_button)
+    }
+
+    @Test
+    fun btnIsCorrectlyDisplayedZone() {
+        Espresso.onView(ViewMatchers.withId(R.id.ic_home))
+            .perform(click())
+        val mockedDb = HelperTestFunction.defaultMockDatabase()
+        currentDatabase = mockedDb
+
+        Espresso.onView(ViewMatchers.withId(R.id.id_zone_management_button))
+            .perform(scrollTo(), click())
+        Espresso.onView(ViewMatchers.withId(R.id.id_new_zone_button))
+            .perform(click())
+
+        Espresso.onView(ViewMatchers.withId(R.id.id_btn_modify_coordinates))
+            .perform(click())
```

Comment:
You can also use Barista and do clickOn(...)

Reply:
Yep, I will use this in later tests since these ones already work :)